tests/monitoring/metrics.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The Kubevirt Authors
+ *
+ */
+
+package monitoring
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/onsi/gomega/types"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	virtapi "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-api"
+	virtcontroller "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-controller"
+	virtoperator "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-operator"
+
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libmonitoring"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-monitoring]Metrics", decorators.SigMonitoring, func() {
+	var virtClient kubecli.KubevirtClient
+	var metrics *libmonitoring.QueryRequestResult
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		setupVM(virtClient)
+		metrics = fetchPrometheusMetrics(virtClient)
+	})
+
+	Context("Prometheus metrics", func() {
+		var excludedMetrics = map[string]bool{
+			// virt-api
+			// can later be added in pre-existing feature tests
+			"kubevirt_portforward_active_tunnels":  true,
+			"kubevirt_usbredir_active_connections": true,
+			"kubevirt_vnc_active_connections":      true,
+			"kubevirt_console_active_connections":  true,
+
+			// virt-controller
+			// needs a migration - ignoring since already tested in - VM Monitoring, VM migration metrics
+			"kubevirt_vmi_migration_phase_transition_time_from_creation_seconds": true,
+			"kubevirt_vmi_migrations_in_pending_phase":                           true,
+			"kubevirt_vmi_migrations_in_scheduling_phase":                        true,
+			"kubevirt_vmi_migrations_in_running_phase":                           true,
+			"kubevirt_vmi_migration_succeeded":                                   true,
+			"kubevirt_vmi_migration_failed":                                      true,
+		}
+
+		It("should contain virt components metrics", func() {
+			metrics = fetchPrometheusMetrics(virtClient)
```

Comment:
sorry for splitted review! Do we need this? this is already performed in `BeforeEach` block

Reply:
no worries
nice catch, I think its a leftover of before I added the 2nd test